src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -159,11 +169,13 @@ public function onKernelResponse(ResponseEvent $event)
 
         $request = $event->getRequest();
 
-        if (!$request->hasSession()) {
+        if (!$request->hasSession() || !$request->attributes->get('_security_firewall_run', false)) {
```

Comment:
Ah, this relates to my question earlier about how the listener is registered.

Reply:
Indeed :)